### PR TITLE
Add license and SourceBuild.props to satisfy post-build checks

### PR DESCRIPTION
**Pull request by crummel**
Since we use code out of the `llvm` directory I copied the license file from there to the root which is where the tooling looks for it.

### CODE REVIEW

**Comment by crummel** [comment]
@leecow Does this seem like a reasonable fix for the licensing issue?  I think the other approach would be having the root license.txt just say "See individual directories for license details" or similar.

**Comment by leecow** [comment]
The file move seems like the right way to go.

File: LICENSE.TXT [hunk 1]
```diff
@@ -0,0 +1,279 @@
+==============================================================================
```
Comment (akoeplinger) [inline LICENSE.TXT]: I'd prefer setting the `LicenseFile` property and pointing it to the original file location

**Comment by akoeplinger** [comment]
ping?

File: LICENSE.TXT (hunk 1 above)
Comment (crummel) [reply LICENSE.TXT]: Does this seem like it will work?  I'll try it out in the source-build context.

File: eng/SourceBuild.props [hunk 2]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <GitHubRepositoryName>llvm-project</GitHubRepositoryName>
     <SourceBuildManagedOnly>true</SourceBuildManagedOnly>
+    <PackageLicenseFile>$(RepoRoot)/llvm/LICENSE.TXT</PackageLicenseFile>
```
Comment (akoeplinger) [inline eng/SourceBuild.props:6]: we're already setting this here: https://github.com/dotnet/llvm-project/blob/7de031a308e84d438576d47d2e24b47c00f3f22f/nuget/packaging.props#L6

I wonder why that isn't working in the SourceBuild case?

I'm fine with checking this in if it unblocks you though.

**Review by akoeplinger — APPROVED** [review]

File: eng/SourceBuild.props (hunk 2 above)
Comment (crummel) [reply eng/SourceBuild.props:6]: Thanks, I'll write up an issue to look into this further but that would be good.

Comment (akoeplinger) [reply eng/SourceBuild.props:6]: Ah I didn't see that this was targetting the `objwriter/12.x` branch which is owned by the JIT team. We've recently merged the objwriter changes into main so for .NET8 there should only be one branch: https://github.com/dotnet/llvm-project/pull/287

Do you need it in earlier branches or can this be retargeted to main?

**Comment by tmds** [comment]
https://github.com/dotnet/llvm-project/pull/439 supersedes this pr. This may be closed.

**Comment by akoeplinger** [comment]
Let's continue this in https://github.com/dotnet/llvm-project/pull/439

**Pull request closed without merging**